KarhooUISDK/Screens/CheckoutScreen/Checkout/KarhooCheckoutPresenter.swift

```diff
@@ -548,32 +541,24 @@ final class KarhooCheckoutPresenter: CheckoutPresenter {
         analytics.checkoutOpened(quote)
     }
 
-    private func reportBookingEvent() {
-        guard let origin = journeyDetails.originLocationDetails else { return }
-        
-        func buildTripForAnalytics() -> TripInfo {
-             TripInfo(
-                origin: origin.toTripLocationDetails(),
-                destination: journeyDetails.destinationLocationDetails?.toTripLocationDetails(),
-                dateScheduled: journeyDetails.scheduledDate,
-                quote: quote.toTripQuote()
-             )
-        }
-        
-        analytics.bookingRequested(tripDetails: trip ?? buildTripForAnalytics())
+    private func reportBookingEvent(quoteId: String) {
+        analytics.bookingRequested(quoteId: quoteId)
     }
 
-    private func reportPaymentSuccess() {
-        analytics.paymentSucceed()
+    private func reportBookingSuccess(tripId: String, quoteId: String?, correlationId: String?) {
+        analytics.bookingSuccess(tripId: tripId, quoteId: quoteId, correlationId: correlationId)
     }
 
-    private func reportPaymentFailure(_ message: String) {
-        analytics.paymentFailed(
-                message: message,
-                last4Digits: retrievePaymentNonce()?.lastFour ?? "",
-                date: Date(),
-                amount: quote.price.highPrice.description,
-                currency: quote.price.currencyCode
+    private func reportBookingFailure(message: String, correlationId: String?) {
+        analytics.bookingFailure(
+            quoteId: quote.id,
+            correlationId: correlationId ?? "",
+            message: message,
+            lastFourDigits: retrievePaymentNonce()?.lastFour ?? "",
+            paymentMethodUsed: String(describing: KarhooUISDKConfigurationProvider.configuration.paymentManager),
```

Comment:
@dianakarh this value can differ depending on the platform (iOS/And). We should be aware of it. And to be honest, it may be tricky to find another way to get actually PSP name in any other way.

Reply:
We could also use `userService.getCurrentUser()?.paymentProvider?.provider.id` and default to the `KarhooUISDKConfigurationProvider.configuration.paymentManager` if needed